packages/renderer/src/lib/preferences/PreferencesRenderingItemFormat.svelte

```diff
@@ -1,387 +1,147 @@
 <script lang="ts">
-import { faXmark } from '@fortawesome/free-solid-svg-icons';
-import Fa from 'svelte-fa/src/fa.svelte';
 import type { IConfigurationPropertyRecordedSchema } from '../../../../main/src/plugin/configuration-registry';
 import { CONFIGURATION_DEFAULT_SCOPE } from '../../../../main/src/plugin/configuration-registry-constants';
 import ErrorMessage from '../ui/ErrorMessage.svelte';
 import Markdown from '../markdown/Markdown.svelte';
 import { getNormalizedDefaultNumberValue } from './Util';
-import Tooltip from '../ui/Tooltip.svelte';
-import Button from '../ui/Button.svelte';
+import BooleanItem from '/@/lib/preferences/item-formats/BooleanItem.svelte';
+import SliderItem from '/@/lib/preferences/item-formats/SliderItem.svelte';
+import NumberItem from '/@/lib/preferences/item-formats/NumberItem.svelte';
+import FileItem from '/@/lib/preferences/item-formats/FileItem.svelte';
+import EnumItem from '/@/lib/preferences/item-formats/EnumItem.svelte';
+import StringItem from '/@/lib/preferences/item-formats/StringItem.svelte';
 
-let invalidEntry = false;
 let invalidText: string | undefined = undefined;
 export let invalidRecord = (_error: string) => {};
 export let validRecord = () => {};
 export let updateResetButtonVisibility = (_recordValue: any) => {};
 export let resetToDefault = false;
 export let enableAutoSave = false;
 
-export let setRecordValue = (_id: string, _value: string) => {};
+export let setRecordValue = (_id: string, _value: string | boolean | number) => {};
 export let enableSlider = false;
 export let record: IConfigurationPropertyRecordedSchema;
 
 let currentRecord: IConfigurationPropertyRecordedSchema;
 let recordUpdateTimeout: NodeJS.Timeout;
 
-let recordValue: any;
+let recordValue: string | boolean | number;
 $: recordValue;
 $: updateResetButtonVisibility?.(recordValue);
-let checkboxValue = false;
+
 $: if (resetToDefault) {
   recordValue = record.type === 'number' ? getNormalizedDefaultNumberValue(record) : record.default;
-  if (typeof recordValue === 'boolean') {
-    checkboxValue = recordValue;
+  if (ensureType(recordValue)) {
+    update(record);
   }
-  update(record);
+
   resetToDefault = false;
 }
 
 $: if (currentRecord !== record) {
   if (record.scope === CONFIGURATION_DEFAULT_SCOPE) {
     if (record.id) {
       window.getConfigurationValue(record.id, record.scope).then(value => {
-        recordValue = value;
+        recordValue = value as string | boolean | number;
```

Comment:
hi, why do we need to do this cast ?

Reply:
Because `value` is typed as `unknown` and before it was not needed since `recordValue` was declared as `any`, and I reduced the scope to `string | boolean | number` to avoi using any.